users/userRouter.js

```diff
@@ -1,47 +1,163 @@
 const express = require('express');
+const users = require('../users/userDb');
+const posts = require('../posts/postRouter');
 
 const router = express.Router();
 
-router.post('/', (req, res) => {
-  // do your magic!
+router.post('/', validateUser(), async (req, res) => {
+  const { newUser } = req.body.name;
+  users.insert({ newUser })
+  .then(newUser => {
+    res.status(201).json(newUser)
+  })
+  .catch(err => {
+    res.status(500).json({
+      errorMessage: "Error saving user to the database."
+    })
+  });
 });
 
-router.post('/:id/posts', (req, res) => {
-  // do your magic!
+router.post('/:id/posts', validateUserId(), validatePost(), (req, res) => {
+  const newPost = {
+    text: req.body.text,
+    user_id: req.params.id
+  }
+
+  posts.insert(newPost)
+    .then(newPost => {
+      res.status(201).json(newPost)
+    })
+    .catch(err => {
+      res.status(500).json({
+        errorMessage: "Error saving the post."
+      })
+    })
 });
 
 router.get('/', (req, res) => {
-  // do your magic!
+  users.get()
+    .then(users => {
+      res.status(200).json(users)
+    })
+    .catch(err => {
+      res.status(500).json({
+        errorMessage: "Error fetching the user database."
+      })
+    })
 });
 
-router.get('/:id', (req, res) => {
-  // do your magic!
+router.get('/:id', validateUserId(), (req, res) => {
```

Comment:
the spec of validateUserId is that is saves user to req.user.  Therefore, you don't need to run a second `getById` on line 52.  You can just return `res.json(req.user)`